website/draft_site/drafts/views.py

```diff
@@ -1,3 +1,185 @@
-from django.shortcuts import render
+import logging
+import os
+import pickle
+from django.shortcuts import render, get_object_or_404
+from django.http import HttpResponse, HttpResponseRedirect
+from django.urls import reverse
+from django.db import transaction
+from django.conf import settings
 
-# Create your views here.
+from . import models
+from mtg_draft_ai.controller import create_packs
+from mtg_draft_ai.api import DraftInfo, Drafter, read_cube_toml
+from mtg_draft_ai.brains import GreedySynergyPicker
+
+
+CUBE_FILE = os.path.join(settings.DRAFTS_APP_DIR, 'cube_81183_tag_data.toml')
+CUBE_LIST = read_cube_toml(CUBE_FILE)
+CARDS_BY_NAME = {c.name: c for c in CUBE_LIST}
+PICKER_FACTORY = GreedySynergyPicker.factory(CUBE_LIST)
+LOGGER = logging.getLogger(__name__)
+
+
+class StaleReadError(Exception):
+    pass
+
+
+# /
+def index(request):
+    return render(request, 'drafts/index.html', {})
+
+
+# /draft
+@transaction.atomic
+def create_draft(request):
+    draft_info = DraftInfo(card_list=CUBE_LIST, num_drafters=6, num_phases=3, cards_per_pack=15)
+    packs = create_packs(draft_info)
+
+    new_draft = models.Draft(num_drafters=draft_info.num_drafters, num_phases=draft_info.num_phases,
+                             cards_per_pack=draft_info.cards_per_pack)
+    new_draft.save()
+
+    for phase in range(0, draft_info.num_phases):
+        for start_seat in range(0, draft_info.num_drafters):
+            pack = packs.get_pack(phase=phase, starting_seat=start_seat)
+            for card in pack:
+                db_card = models.Card(draft=new_draft, name=card.name, phase=phase, start_seat=start_seat)
+                db_card.save()
+
+    # Create human drafter in DB
+    models.Drafter(draft=new_draft, seat=0, bot=False).save()
+    # Create bot drafters in DB
+    # Seat 0 is the human drafter, so start at seat 1
+    for i in range(1, draft_info.num_drafters):
+        # TODO: for now picker state isn't saved to improve performance; we might need to in the future.
+        bot_drafter = Drafter(None, draft_info)
+        db_bot_drafter = models.Drafter(draft=new_draft, seat=i, bot=True, bot_state=pickle.dumps(bot_drafter))
+        db_bot_drafter.save()
+
+    return HttpResponseRedirect(reverse('show_draft', args=[new_draft.id]))
+
+
+# /draft/<int:draft_id>
+def show_draft(request, draft_id):
+    draft = get_object_or_404(models.Draft, pk=draft_id)
```

Comment:
It seems like with multiple calls to get_object_or_404, it could become difficult to troubleshoot errors in our data. They don't appear to log or give us any information as far as what went wrong.

Reply:
Django actually handles this fine on the back-end by throwing an exception and then logging it.

Reply:
We just tried this out and it looks like Django gives enough debug info for the 404 by default, since DEBUG = True right now in settings.